deis-cleanup.rb

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+
+require "open3"
+require "csv"
+
+KEEP_LAST_VERSIONS = ENV["KEEP_LAST_VERSIONS"].to_i || 2
+DRY_RUN = ENV["DRY_RUN"] || false
+EXCLUDE_REGEX = case ENV["EXCLUDE_REGEX"]
+                when nil then /alpine|deis|blackhole|none|datadog|cleanup|heroku|python/
+                else /#{ENV["EXCLUDE_REGEX"]}/
+                end
+
+def images
+  output, status = Open3.capture2e("docker images")
+  abort output unless status.success?
+
+  headers, *rows = CSV.parse(output, col_sep: " ").map { |e| e.first(2) }
+  headers.map! { |header| header.downcase.to_sym }
+  rows.map { |row| headers.zip(row).to_h }
+end
+
+excluded = -> (image) { image[:repository].match EXCLUDE_REGEX }
+image_type = -> (image) { image[:tag] =~ /git/ ? :git : :versioned }
+repository_name = -> (image) { image[:repository] }
```

Comment:
`image_type` and `excluded` are extracted to lambdas due to readability. `filtered` assignment will exceed 120 chars otherwise.